### PR TITLE
Add type annotations to a few files

### PR DESCRIPTION
**Pull request by qwergle**
This PR just adds some type annotations to a few files, and is meant to help with fixing #892. It's not much, but I'm still working on adding type annotations to all files.

### CODE REVIEW

File: bpython/args.py [hunk 1]
```diff
@@ -62,11 +63,11 @@ def version_banner(base="bpython"):
     )
 
 
-def copyright_banner():
+def copyright_banner() -> str:
     return _("{} See AUTHORS.rst for details.").format(__copyright__)
 
 
-def parse(args, extras=None, ignore_stdin=False):
+def parse(args, extras=None, ignore_stdin=False) -> Tuple:
```
Comment (sebastinas) [inline bpython/args.py:70]: This could list the types of the tuple's elements.

File: bpython/autocomplete.py [hunk 2]
```diff
@@ -207,11 +208,11 @@ def method_match_fuzzy(word, size, text):
 class BaseCompletionType:
     """Describes different completion types"""
 
-    def __init__(self, shown_before_tab=True, mode=AutocompleteModes.SIMPLE):
+    def __init__(self, shown_before_tab: bool=True, mode=AutocompleteModes.SIMPLE) -> None:
```
Comment (sebastinas) [inline bpython/autocomplete.py:211]: mode could be annotated here as well.

File: bpython/autocomplete.py [hunk 3]
```diff
@@ -207,11 +208,11 @@ def method_match_fuzzy(word, size, text):
...
         self._shown_before_tab = shown_before_tab
         self.method_match = MODES_MAP[mode]
 
-    def matches(self, cursor_offset, line, **kwargs):
+    def matches(self, cursor_offset, line, **kwargs) -> NoReturn:
```
Comment (sebastinas) [inline bpython/autocomplete.py:215]: Is this a good idea for the definition in the base class?

**Review by sebastinas — CHANGES_REQUESTED** [review]
Please also run `black` to ensure consistent formatting after your changes.

**Comment by supremestdoggo** [comment]
The commit I just added performs all of the changes you requested, as well as reformatting many files using black.

**Comment by sebastinas** [comment]
hmm, I had to rerun black again. Merged after doing that.

**Pull request closed without merging**